### PR TITLE
Remove authorization from specific routes

### LINKED ISSUE #569: Add authorization to specific routes
#### Nature of issue?

- [X] Existing feature enhancement

#### Feature enhancement details:
Certain routes should not be accessible by all users! One that comes to mind is a user's settings (`/<username>/settings`) and a user's assets (`/<username/assets`). Maybe these routes should not include the username and just use the user that it stored in the cookie? Or they should redirect to `/` if a user who is not logged in as the `<username>` tries to navigate to one of these routes. The former appears to be what most websites do.

### PR DESCRIPTION
**Pull request by shinytang6**
Before your pull request is reviewed and merged, please ensure that:

* [x] there are no linting errors -- `npm run lint`
* [x] your code is in a uniquely-named feature branch and has been rebased on top of the latest master. If you're asked to make more changes, make sure you rebase onto master then too!
* [x] your pull request is descriptively named and links to an issue number, i.e. `Fixes #123`

Thank you! Fixes #569

### CODE REVIEW

File: server/controllers/aws.controller.js [hunk 1]
```diff
@@ -108,6 +108,11 @@ export function copyObjectInS3(req, res) {
   });
 }
 
+export function listObjectsInS3(req, res) {
+  req.params = Object.assign({}, req.user);
```
Comment (catarak) [inline server/controllers/aws.controller.js]: i think overwriting req.params is confusing—maybe `listObjectsInS3ForUser` could check if req.params.username is defined, and if not, use req.user?

**Comment by catarak** [comment]
woohoo thanks for this! sorry about the travis build fail, it's related to #640.

**Comment by shinytang6** [comment]
@catarak l am thinking that maybe `/:username/sketches` should also be removed, what do you think?

**Comment by catarak** [comment]
@shinytang6 we should keep that as it's currently the only way to see another user's sketches.

File: server/controllers/aws.controller.js (hunk 1 above)
Comment (shinytang6) [reply server/controllers/aws.controller.js]: done :)

**Comment by catarak** [comment]
for this PR, you also need to change the routes, client-side, and change the routes in `client/components/Nav.jsx`.

**Comment by shinytang6** [comment]
Do you mean removing all `/:username/assets` and `/:username/account` on client side? l think there's no need to remove that cause they are only valid when users logged in click 'My assets/settings'

**Comment by catarak** [comment]
Yes, those routes should be moved to `/assets` and `/account` client side, and the `<Link>`s should be changed in the navigation as well.

File: server/routes/server.routes.js [hunk 2]
```diff
@@ -57,7 +57,15 @@ router.get('/verify', (req, res) => {
 });
 
 router.get('/sketches', (req, res) => {
-  res.send(renderIndex());
+  req.user ? res.send(renderIndex()) : get404Sketch(html => res.send(html));
+});
+
+router.get('/assets', (req, res) => {
+  req.user ? res.send(renderIndex()) : get404Sketch(html => res.send(html));
```
Comment (catarak) [inline server/routes/server.routes.js]: After using this, I think it's confusing that this renders a 404 page if the user is not logged in—I think this should redirect to `/login`.

File: server/routes/server.routes.js [hunk 3]
```diff
@@ -57,7 +57,15 @@ router.get('/verify', (req, res) => {
...
+});
+
+router.get('/account', (req, res) => {
+  req.user ? res.send(renderIndex()) : get404Sketch(html => res.send(html));
```
Comment (catarak) [inline server/routes/server.routes.js]: After using this, I think it's confusing that this renders a 404 page if the user is not logged in—I think this should redirect to `/login`.

File: server/routes/server.routes.js [hunk 4]
```diff
@@ -57,7 +57,15 @@ router.get('/verify', (req, res) => {
 });
 
 router.get('/sketches', (req, res) => {
-  res.send(renderIndex());
+  req.user ? res.send(renderIndex()) : get404Sketch(html => res.send(html));
```
Comment (catarak) [inline server/routes/server.routes.js]: This should also redirect to login rather than be a 404!

**Comment by catarak** [comment]
This is working great! Just one last thing to fix.

**Pull request merged**